subs/pantry/src/Pantry/Storage.hs

```diff
@@ -215,15 +215,17 @@ initStorage
 initStorage fp inner = do
   ensureDir $ parent fp
   bracket
-    (createSqlitePoolFromInfo sqinfo 1)
+    (createSqlitePoolFromInfo (sqinfo False) 1)
     (liftIO . destroyAllResources) $ \pool -> do
-
-    migrates <- runSqlPool (runMigrationSilent migrateAll) pool
-    forM_ migrates $ \mig -> logDebug $ "Migration output: " <> display mig
-    inner (P.Storage pool)
+    migrates <- wrapMigrationFailure $ runSqlPool (runMigrationSilent migrateAll) pool
+    forM_ migrates $ \mig -> logDebug $ "Migration executed: " <> display mig
+  bracket
+    (createSqlitePoolFromInfo (sqinfo True) 1)
+    (liftIO . destroyAllResources) $ \pool -> inner (P.Storage pool)
   where
-    sqinfo = set extraPragmas ["PRAGMA busy_timeout=2000;"]
-           $ set fkEnabled True
+    wrapMigrationFailure = handle (throwIO . MigrationFailure)
```

Comment:
How about including the `FilePath` above in the `MigrationFailure` data constructor, and replacing the `$STACK_ROOT` bit with the actual file path they would need to delete?

Reply:
Fixed